### ISSUE TITLE
Live broadcast streaming delay

### ISSUE THREAD

**Issue by guitarpound** [arut/nginx-rtmp-module#1499]
Describe the question
Hi,

I setup the AVideo Server in LAN and then do live streaming (RTMP).

The problem is when I compare Video streaming output with the video source.
It delay about 35-40 sec.
Is it possible to reduce the latency time close to video source?

Thanks & BR,

**Comment by sriman** [comment]
Yes we can achieve 2 seconds sub latency

**Comment by guitarpound** [comment]
Hi Sriman,

So, It depend on the Encoder setting.  Right?
I tried set the parameter same as your picture.  But the latency still about 10-12 sec.

![obs](https://user-images.githubusercontent.com/57282919/79033648-20b45e80-7bda-11ea-9d2a-6d1ed77e94a9.png)

Thanks & BR,
Pond

**Comment by sriman** [comment]
No other than this I haven't configured anything, I have served through servers in which I could achieve 1.5 - 2 seconds delay, you are using LAN so it should be below 1 second.

**Comment by ArvyRogerio** [comment]
> So, It depend on the Encoder setting. Right?

Not only. Screen size, encoding, decoding, internet speed + latency, etc. I use Nginx + RTMP + OBS, 1280px video, 128k audio, remote server, my latency is exactly 40 seconds, always.

**Comment by limbo123** [comment]
> > So, It depend on the Encoder setting. Right?
> 
> Not only. Screen size, encoding, decoding, internet speed + latency, etc. I use Nginx + RTMP + OBS, 1280px video, 128k audio, remote server, my latency is exactly 40 seconds, always.

Did you figure it out? I have exact same issue, 40 seconds delay and I'm also converting rtmp to hls, here's my nginx configuration (changing the fragment and playlist length is useless in my case, the delay remains the same):

```
events {}

rtmp {
    server {
        listen 1935;

        application live {
            live on;
            hls on;
            hls_path /tmp/hls;
            hls_fragment 5s;
            hls_playlist_length 15s;          
            hls_base_url http://localhost:8080/hls/;
            on_publish http://backend:3000/streams/start;
            on_done http://backend:3000/streams/stop;
        }
    }
}

http {
    server {
        listen 8080;

        location /hls {
            types {
                application/vnd.apple.mpegurl m3u8;
                application/octet-stream ts;
            }
            root /tmp;
            add_header Cache-Control no-cache;

            add_header Access-Control-Allow-Origin *;
        }
    }
}
```

**Comment by guitarpound** [comment]
> > > So, It depend on the Encoder setting. Right?
> > 
> > 
> > Not only. Screen size, encoding, decoding, internet speed + latency, etc. I use Nginx + RTMP + OBS, 1280px video, 128k audio, remote server, my latency is exactly 40 seconds, always.
> 
> Did you figure it out? I have exact same issue, 40 seconds delay and I'm also converting rtmp to hls, here's my nginx configuration (changing the fragment and playlist length is useless in my case, the delay remains the same):
> 
> ```
> events {}
> 
> rtmp {
>     server {
>         listen 1935;
> 
>         application live {
>             live on;
>             hls on;
>             hls_path /tmp/hls;
>             hls_fragment 5s;
>             hls_playlist_length 15s;          
>             hls_base_url http://localhost:8080/hls/;
>             on_publish http://backend:3000/streams/start;
>             on_done http://backend:3000/streams/stop;
>         }
>     }
> }
> 
> http {
>     server {
>         listen 8080;
> 
>         location /hls {
>             types {
>                 application/vnd.apple.mpegurl m3u8;
>                 application/octet-stream ts;
>             }
>             root /tmp;
>             add_header Cache-Control no-cache;
> 
>             add_header Access-Control-Allow-Origin *;
>         }
>     }
> }
> ```

You can try add below command in nginx config 
And restart nginx

rtmp {
server {
listen 1935;
chunk_size 4096;

```
    application live {
        live on;
        sync 1s;
        drop_idle_publisher 5s;
        hls on;
        hls_path /HLS/live;
        hls_fragment 1s;
        hls_playlist_length 3s;
        idle_streams off;
    }
}
```

}

http {

tcp_nodelay on;

```
server {
    listen 8080;
    server_name localhost;
    …..
    …..
}
```
}

**Comment by ArvyRogerio** [comment]
No, after some time I moved to a SRT server solution: https://github.com/Edward-Wu/srt-live-server

> > Not only. Screen size, encoding, decoding, internet speed + latency, etc. I use Nginx + RTMP + OBS, 1280px video, 128k audio, remote server, my latency is exactly 40 seconds, always.
> 
> Did you figure it out? I have exact same issue, 40 seconds delay and I'm also converting rtmp to hls, here's my nginx configuration (changing the fragment and playlist length is useless in my case, the delay remains the same):
